app/web/features/dashboard/Hero/HeroButton.tsx

```diff
@@ -0,0 +1,65 @@
+import { createTheme, Theme, ThemeProvider, useTheme } from "@material-ui/core";
+import Button from "components/Button";
+import { DASHBOARD } from "i18n/namespaces";
+import Link from "next/link";
+import { useTranslation } from "next-i18next";
+import { useMemo } from "react";
+import { searchRoute } from "routes";
+import makeStyles from "utils/makeStyles";
+
+// because this component is over an image background and has a special button, we adjust the theme
+const getHeroButtonTheme = (theme: Theme) =>
```

Comment:
Any particular reason to use a theme and not `className`? We can target inner elements if necessary with the relevant mui classes

Reply:
Yeah, I've opted to set a custom theme. My reasoning is that changing the color of the button feels like an override of an internal behavior of the library (applying the primary color on the primary button backgrounds), and we don't really know what the library does or will do in the future. Like maybe MUI in future releases will make the hover color be a darkened version of the button color, like Charka UI does. So either you know the internals of the library and override all the necessary CSS rules, or you pass a custom theme and don't worry about how the library sets or manipulates the theme colors when they are used on buttons or other components.

Your comment is a good prompt though, because reviewing this, it feels a bit strange to add a custom theme just for 1 button, when the entire hero block has also different styling needs.

So I've given it a pass in [this commit](https://github.com/Couchers-org/couchers/pull/2918/commits/2e3a42da41a4a4d36d79243d15b025b8a8ee7d3f), making a single new theme for the Hero block contents that are over the background image, and applying this to all the components over it. 

If we change the background image, and these colors should change, we can just change this theme.

Let me know your thoughts! :)

Reply:
I don't feel strongly about it, so feel free to leave it as is, but to me we are indeed overriding MUI behaviour so the className targeting/ is appropriate (although I take your point about MUI using the theme colours in non-obvious ways).

Using `<ThemeProvider>` this way doesn't feel very theme-like - themes are a style that apply to the whole page or large sections, having a theme for a single button or something isn't what a theme is for. It would make sense if it was applied to the whole Hero section, but then obviously all the text would be white etc.

Reply:
Right, I thought about that too originally. It's possible, yes. But then, I'd make another `ThemeProvider` inside for the search box, to set different theme styles for all the components within it. Either way works.

<img width="570" alt="Screen Shot 2022-05-18 at 11 30 43" src="https://user-images.githubusercontent.com/1557348/169007422-51394377-993d-45ee-b940-af29770b7c3f.png">

About nesting themes, if it helps I can point to the docs where it's referenced as a standard pattern for this type of scenario: https://mui.com/material-ui/customization/theming/#nesting-the-theme

> You can nest multiple theme providers. This can be really useful when dealing with different areas of your application that have distinct appearance from each other.